### PR TITLE
Fix for #400

### LINKED ISSUE #400: [New Feature] Allow users to rotate pages of an existing PDF
- `Rotate Pages` :  Add a new option in the dialog in `View Files` (The dialog opens up when we select a PDF)
- On selecting this option, open up a dialog asking user the angle by which they want to rotate their pages.
- On selecting `ok`, save the PDF with name `<pdf_name>_rotated<angle>.pdf`

### Reference:
https://developers.itextpdf.com/question/how-rotate-page-90-degrees

### PR DESCRIPTION
**Pull request by vyankatesh24**
# Description
Added an option to rotate pages in existing PDFs.

image

image

Fixes #400 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

### CODE REVIEW

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 1]
```diff
@@ -150,9 +157,51 @@ private void performOperation(int index, int position, File file) {
...
                 mPDFEncryptionUtils.removePassword(file.getPath(), ViewFilesAdapter.this, mFileList);
                 break;
+
+            case 8://Rotate Pages
+                new MaterialDialog.Builder(mActivity)
+                        .title(R.string.rotate_pages)
+                        .content(R.string.enter_rotation_angle)
+                        .input(null, null, false, (dialog, input) -> {
+                            String sourceFilePath = file.getPath();
+                            String destFilePath = sourceFilePath.substring(0, sourceFilePath.lastIndexOf('/') + 1);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: add a function in FileUtils to get just the path

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 2]
```diff
@@ -150,9 +157,51 @@ private void performOperation(int index, int position, File file) {
...
+                                        page.put(PdfName.ROTATE, new PdfNumber(angle));
+                                    } else {
+                                        page.put(PdfName.ROTATE, new PdfNumber((rotate.intValue() + angle) % 360));
+                                    }
+                                }
+                                PdfStamper stamper = new PdfStamper(reader, new FileOutputStream(destFilePath));
+                                stamper.close();
+                                reader.close();
+                                showSnackbar(mActivity, R.string.snackbar_pdfCreated);
+                                notifyDataSetChanged();
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: add file in `mFileList`, else `notifyDataSetChanged()` won't add it to the list

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 3]
```diff
@@ -150,9 +157,51 @@ private void performOperation(int index, int position, File file) {
...
+                        .title(R.string.rotate_pages)
+                        .content(R.string.enter_rotation_angle)
+                        .input(null, null, false, (dialog, input) -> {
+                            String sourceFilePath = file.getPath();
+                            String destFilePath = sourceFilePath.substring(0, sourceFilePath.lastIndexOf('/') + 1);
+                            String fileName = sourceFilePath.substring(sourceFilePath.lastIndexOf('/') + 1);
+                            destFilePath += String.format(mActivity.getString(R.string.rotated_file_name),
+                                    fileName.substring(0, fileName.lastIndexOf('.')), input,
+                                    mActivity.getString(R.string.pdf_ext));
+                            try {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Add this function in `PDFUtils.java`

Comment (vyankatesh24) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Whole code or just rotation code?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: just the rotation code.

Just send source & destination filename to that function.

**Pull request merged**